api/src/main/java/com/epam/pipeline/manager/datastorage/DataStorageManager.java

```diff
@@ -343,6 +353,40 @@ public AbstractDataStorage updatePolicy(DataStorageVO dataStorageVO) {
         return createdStorage;
     }
 
+    @Transactional(propagation = Propagation.REQUIRED)
+    public AbstractDataStorage createDefaultStorageForUser(final String userName) {
+        final String dataStorageTemplateJson =
```

Comment:
To enable json validation on the GUI, we need to change this preference to `ObjectPreference` type. So handling of preference in code will change - we need to get it string value ta first and then proceed with replacement, etc. Do not forget to handle `null` value.

Reply:
Done, if no template specified, build default DataStorageTemplate based on a user's name